lang/cs/Org.Apache.REEF.Network/Group/Operators/IGroupCommOperator.cs

```diff
@@ -39,5 +39,10 @@ public interface IGroupCommOperator<T>
         /// The operator version number.
         /// </summary>
         int Version { get; }
+
+        /// <summary>
+        /// Ensure all parent and children nodes in the topology are registered with teh Name Service.
+        /// </summary>
+        void WaitForRegistration();
```

Comment:
The sequence must be
1. Complete all nested constructors to ensure handlers/observers are registered 
2. Then in GroupCommClient, Register with name service
3. Then call WaitForRegistration in each CommunicationGroupClient and each Operator. 

The problem was the operators is Dictionary<string, object>.  We Know the object is IGroupCommOperator of T, but we don't know T given the dictionary of operators. That was the reason I didn't call it in GroupCommClient constructor. 

However, I can use reflection to do it. The code change will follow.